pkg/util/test/util.go

```diff
@@ -4160,6 +4162,119 @@ func ValidateAlertManagerDisabled(pxImageList map[string]string, cluster *corev1
+
+	isCOenabled := strings.Contains(configMap.Data["config_properties_px.yaml"], "certStoreType: \"kvstore\"")
+	if !isCOenabled {
+		return fmt.Errorf("telemetry container orchestrator 'certStoreType: kvstore' is expected to be set in 'px-telemetry-register' configmap")
+	}
+
+	logrus.Infof("Validated 'certStoreType: kvstore' configmap setting")
+
+	// Test validation is done by checking timestamp entries of when the container orchestrator operations were called (server start, secret set/get).
+	// The timestamps for secret set/get must be after the timestamp for the start.
+	labelSelector := map[string]string{"role": "px-telemetry-registration"}
+	nodeList, err := coreops.Instance().GetNodes()
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("find telemetry registration pod node and px pod on that node...")
+
+	var pxPod v1.Pod
+	for _, node := range nodeList.Items {
+		// Get telemetry registration pod
+		podList, err := coreops.Instance().GetPodsByNodeAndLabels(node.Name, cluster.Namespace, labelSelector)
+		if err != nil {
+			return err
+		}
+
+		if len(podList.Items) == 0 {
+			continue
+		}
+		tregPod := podList.Items[0]
+		logrus.Infof("Found telemetry registration pod [%s] on node [%s]", tregPod.Name, node.Name)
+		labelSelector = map[string]string{"name": "portworx"}
+		podList, err = coreops.Instance().GetPodsByNodeAndLabels(node.Name, cluster.Namespace, labelSelector)
+		if err != nil {
+			return err
+		}
+
+		if len(podList.Items) == 0 {
+			return fmt.Errorf("failed to find Portworx pod on telemetry registration node [%s]", node.Name)
+		}
+		pxPod = podList.Items[0]
+		logrus.Infof("Found PX pod [%s] on telemetry registration node [%s]", pxPod.Name, node.Name)
+		break
+	}
+
+	getCoStateTimeStamp := func(tsFileName string) (int64, error) {
+		tmData, err := runCmdInsidePxPod(&pxPod, "cat "+tsFileName, cluster.Namespace, false)
+		if err != nil {
+			return 0, err
+		}
+
+		tm, terr := strconv.ParseInt(string(tmData), 10, 64)
+		if terr != nil {
+			return 0, terr
+		}
+		return tm, nil
+	}
+
+	coStartTime, startErr := getCoStateTimeStamp(coStateDir + "/kvStart.ts")
+	if startErr != nil {
+		logrus.Errorf("error obtaining start time from costate file: %v\n", startErr)
+		return fmt.Errorf("container orchestrator validated failed, error obtaining start time from costate file")
+	}
+	logrus.Infof("container orchestrater server start time: %v", time.Unix(coStartTime, 0))
+
+	coSetTime, setErr := getCoStateTimeStamp(coStateDir + "/kvSet.ts")
+	coGetTime, getErr := getCoStateTimeStamp(coStateDir + "/kvGet.ts")
+
+	if setErr == nil {
+		// secret set time exists, check it
+		logrus.Infof("container orchestrater set time: %v\n", time.Unix(coSetTime, 0))
+		if coSetTime > coStartTime {
+			logrus.Infof("container orchestrator validated via set time")
+			return nil
+		}
+		setErr = fmt.Errorf("set time is less than start time, check for upgrade")
+		logrus.Infof("%s", setErr.Error())
```

Comment:
Is this error meant to be just info log? Can maybe just be printed as `logrus.Info("set time is less than start time, check for upgrade")`?

Reply:
This error is output as a log message,  but the error is created so it can be checked in the next if condition.